chain/messagepool/messagepool.go

```diff
@@ -863,7 +847,20 @@ func (mp *MessagePool) checkBalance(ctx context.Context, m *types.SignedMessage,
 	return nil
 }
 
-func (mp *MessagePool) addTs(ctx context.Context, m *types.SignedMessage, curTs *types.TipSet, local, untrusted bool) (bool, error) {
+func (mp *MessagePool) addTs(ctx context.Context, m *types.SignedMessage, local, untrusted bool) (bool, error) {
+	//ensures that we have a consistent view of the state
+	mp.transactionLk.Lock()
+	defer mp.transactionLk.Unlock()
+
+	mp.stateLk.RLock()
+	err := mp.checkMessage(ctx, m)
```

Comment:
I think we can refactor `checkMessage` to not need to reference `curTs` at all -- I can push up a commit that does that, and I think in that case we can change this to only take the `stateLk` _after_ having called `checkMessage`.

Reply:
That sounds great to me! Have you gotten to that yet? Should we set up a new ticket for that?